src/Frame.cc

```
@@ -1840,7 +1840,8 @@ bool Frame::ExportCASAImage(casacore::ImageInterface<casacore::Float>& image, fs
     bool success(false);
 
     // Remove the old image file if it has a same file name
-    if (fs::exists(output_filename)) {
+    std::error_code error_code;
+    if (fs::exists(output_filename, error_code) && !error_code) {
```

Why do you check twice the result of `fs::exists`? The return value is `true` if the path exists and `false` if it does not. You can use `error_code` just in case something goes wrong. I suggest you change this line and use `error_code` afterward only to investigate what went wrong (a system-dependent error code). Do you really need it?

---

This comment propagates to the rest of the changes, for both cases:
```
fs::exists(p, ec) && !ec
```
and
```
!fs::exists(p, ec) || ec
```

---

I suggest adding across the changes something like:
```
if (fs::exists(p, ec) {
    // ...
} else {
    spdlog::debug(fmt::format("failed with error code {}", ec));
}
```
or just ignore the error_code result.

---

I think we can just ignore the error code, you're right. It's really just there so that we can use the `no_except` version of the call